File: lint-arwaky/src-python/contract/lint_reporting_protocol.py
"""lint_reporting_protocol — Protocols for report formatting and output.

Defines interfaces for transforming GovernanceReports into various output formats.
"""

from abc import ABC, abstractmethod

from ..taxonomy import FileFormat, GovernanceReport, LogOutput, ResponseData


class ILintReportFormatterProtocol(ABC):
    """Protocol for formatting lint reports into various representations."""

    @abstractmethod
    def format(self, report: GovernanceReport) -> LogOutput:
        """Format the report as a plain text representation."""
        ...

    @abstractmethod
    def get_formatted_payload(
        self, report: GovernanceReport, output_format: FileFormat
    ) -> ResponseData:
        """Return formatted data for surface consumption."""
        ...

    @abstractmethod
    def to_sarif(self, report: GovernanceReport) -> LogOutput:
        """Convert report to SARIF format."""
        ...

    @abstractmethod
    def to_junit(self, report: GovernanceReport) -> LogOutput:
        """Convert report to JUnit XML format."""
        ...

File: lint-arwaky/src-rust/contract/lint_reporting_protocol.rs
// lint_reporting_protocol — Protocols for report formatting and output.
// Defines interfaces for transforming GovernanceReports into various output formats.
use crate::taxonomy::{FileFormat, GovernanceReport, LogOutput, ResponseData};
use async_trait::async_trait;

#[async_trait]
pub trait ILintReportingProtocol: Send + Sync {
    /// Format the report as a plain text representation.
    async fn format(&self, report: &GovernanceReport) -> LogOutput;

    /// Return formatted data for surface consumption.
    async fn get_formatted_payload(
        &self,
        report: &GovernanceReport,
        output_format: FileFormat,
    ) -> ResponseData;

    /// Convert report to SARIF format.
    async fn to_sarif(&self, report: &GovernanceReport) -> LogOutput;

    /// Convert report to JUnit XML format.
    async fn to_junit(&self, report: &GovernanceReport) -> LogOutput;
}
